crates/rwasm/machine/src/syscall/fat_op/table/mod.rs

```diff
@@ -31,31 +31,145 @@ where
         let local: &TableInitCols<AB::Var> = (*local).borrow();
         let next: &TableInitCols<AB::Var> = (*next).borrow();
 
+        let is_real = local.is_table_init + local.is_table_fill;
+        let next_is_real = next.is_table_init + next.is_table_fill;
+
+        self.eval_boolean_constraints(builder, local, is_real.clone());
+
+        // First Row Constraints
+        // The first row in the trace must be marked as first
+        builder.when_first_row().assert_one(local.is_first);
+
+        self.eval_real_row_constraints(builder, local, is_real.clone());
+        self.eval_operation_type_constraints(builder, local);
+        self.eval_length_validation_constraints(builder, local);
+
+        // Event Transition Constraints
+        // Between events: if last row and next is real, then next must be first
+        builder
+            .when_transition()
+            .when(local.is_last)
+            .when(next_is_real.clone())
+            .assert_one(next.is_first);
+
+        self.eval_within_event_transition_constraints(
+            builder,
+            local,
+            next,
+            is_real.clone(),
+            next_is_real,
+        );
+        self.eval_value_propagation_constraints(builder, local);
+        self.eval_address_boundary_constraints(builder, local);
+        self.eval_address_increment_constraints(builder, local, next, is_real.clone());
+        self.eval_initial_address_constraints(builder, local);
+        self.eval_range_check_constraints(builder, local);
+        self.eval_memory_access(local, builder);
+        self.eval_syscall_interactions(builder, local);
+    }
+}
+
+impl TableInitFillChip {
+    // Boolean Constraints
+    fn eval_boolean_constraints<AB: SP1AirBuilder>(
+        &self,
+        builder: &mut AB,
+        local: &TableInitCols<AB::Var>,
+        is_real: AB::Expr,
+    ) {
+        // Ensure all flag fields are boolean (0 or 1)
+        builder.assert_bool(local.is_table_init);
+        builder.assert_bool(local.is_table_fill);
         builder.assert_bool(local.is_first);
```

Comment:
why not use is_table_fill_first and is_table_init_first to place is_first everywhere?